HW_11/task_4.py

```diff
@@ -0,0 +1,72 @@
+# Создайте класс Матрица. Добавьте методы для: вывода на печать, проверку на равенство, сложения, *умножения матриц
+class Matrix:
+
+    def __init__(self, matrix: list[list[int]]):
+        self._rows = len(matrix)
+        self._cols = len(matrix[0])
+        self._matrix = matrix
+
+    def __add__(self, other):
+        """Calculates the sum of matrices"""
+        if not isinstance(other, self.__class__):
+            raise TypeError("The object does not belong to the Matrix class")
+        if self._cols != other._cols or self._rows != other._rows:
+            raise ValueError('The operation cannot be performed for matrices of different dimensions')
+        sum_matrix = [[0 for _ in range(self._cols)] for _ in range(self._rows)]
+        for i in range(self._rows):
+            for j in range(self._cols):
+                sum_matrix[i][j] = self._matrix[i][j] + other._matrix[i][j]
+        return Matrix(sum_matrix)
+
+    def __mul__(self, other):
+        """Calculates a multiplication of matrices"""
+        if not isinstance(other, self.__class__):
+            raise TypeError("Not a 'Matrix'-type object")
+        if self._cols != other._rows:
+            raise ValueError(
+                "The number of rows of the first matrix is not equal to the number of columns of the second matrix")
+        mult_matrix = [[0 for _ in range(other._cols)] for _ in range(self._rows)]
+        for i in range(self._rows):
+            for j in range(other._cols):
+                for k in range(self._cols):
+                    mult_matrix[i][j] += self._matrix[i][k] * other._matrix[k][j]
+        return Matrix(mult_matrix)
+
+    def __eq__(self, other) -> bool:
+        """Returns True if the matrix equals to other one"""
+        if self is other:
+            return True
+        if not isinstance(other, self.__class__):
+            raise TypeError("Not a 'Matrix'-type object")
+        if self._rows != other._rows or self._cols != other._cols:
+            return False
+        for j in range(self._rows):
+            for i in range(self._cols):
+                if self._matrix[j][i] != other._matrix[j][i]:
+                    return False
+        return True
```

Comment:
Если ваши матрицы поставляется из себя список списков, то не нужно такой сложной проверки. Оператор равенства вполне подойдет.
```suggestion
    def __eq__(self, other) -> bool:
        """Returns True if the matrix equals to other one"""
        return self._matrix == other._matrix
```